omisego-sdk/src/main/java/co/omisego/omisego/model/transaction/request/TransactionRequestType.kt

```diff
@@ -0,0 +1,24 @@
+package co.omisego.omisego.model.transaction.request
+
+/*
+ * OmiseGO
+ *
+ * Created by Phuchit Sirimongkolsathien on 27/3/2018 AD.
+ * Copyright © 2017-2018 OmiseGO. All rights reserved.
+ */
+
+import co.omisego.omisego.constant.enums.OMGEnum
+import co.omisego.omisego.custom.gson.OMGEnumAdapter
+import com.google.gson.annotations.JsonAdapter
+
+/**
+ * The different types of request that can be generated
+ */
+@JsonAdapter(OMGEnumAdapter::class)
```

Comment:
I've already tried this when I created the `OMGEnumAdapter` and it didn't work 😞

Reply:
😱 You tried by adding the enum on the supertype?
I think it could be done via [https://static.javadoc.io/com.google.code.gson/gson/2.8.2/com/google/gson/GsonBuilder.html#registerTypeAdapterFactory-com.google.gson.TypeAdapterFactory-](registerTypeAdapterFactory) or [https://static.javadoc.io/com.google.code.gson/gson/2.8.2/com/google/gson/GsonBuilder.html#registerTypeHierarchyAdapter-java.lang.Class-java.lang.Object-](registerTypeHierarchyAdapter) (probably the latter)

I'll check when I have some time if you want, nothing urgent, would definitely be a plus to abstract the gson logic from any data class :)

Reply:
Yes 😂.

I've tried `registerTypeHierarchyAdapter`, and it works 👍

Reply:
Awesome! 😄 🎉